cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/arcgis/model/Feature.java

```diff
@@ -309,13 +323,21 @@ public static Feature createInstanceFromJson(String json) throws ArcgisException
      */
     public static Feature createInstanceFromJson(JsonObject json) throws ArcgisException {
         try {
-            Geometry geometry;
+            Geometry geometry = null;
             if (json.has(GEOMETRY_TAG)) {
-                JsonObject jsonGeometry = json.get(GEOMETRY_TAG).getAsJsonObject();
-                geometry = Point.createInstanceFromJson(jsonGeometry); // TODO another 
-                                                                       //geometry types?
-            } else {
-                geometry = null;
+                JsonElement jsonGeometryElement = json.get(GEOMETRY_TAG);
+                if (jsonGeometryElement.isJsonObject()) {
+                    JsonObject jsonGeometry = jsonGeometryElement.getAsJsonObject();
+                    if (jsonGeometry.get("x") != null) {
+                        geometry = Point.createInstanceFromJson(jsonGeometry);
+                    } else if (jsonGeometry.get("paths") != null) {
+                        geometry = PolyLine.createInstanceFromJson(jsonGeometry);
+                    // } else if (jsonGeometry.get("points") != null) {
+                    //     // geometry = MultiPoint.createInstance(jsonGeometry);
+                    // } else if (jsonGeometry.get("rings") != null) {
+                    //     // geometry = Polygon.createInstance(jsonGeometry);
```

Comment:
This leftover should be removed? Or is a kind of placeholder for future extensibility? In the second case, I'd suggest to add a `FIXME` comment telling so.

Reply:
Done in https://github.com/telefonicaid/fiware-cygnus/pull/2394/commits/828d9399a2cbfb0a48540e8169d523af063f1b4c